plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1275,12 +1275,24 @@ private static String validateAvroSchemaLiteral(String avroSchemaLiteral)
 
     private static Location getValidatedExternalLocation(String location)
     {
+        Location validated;
         try {
-            return Location.of(location);
+            validated = Location.of(location);
         }
         catch (IllegalArgumentException e) {
             throw new TrinoException(INVALID_TABLE_PROPERTY, "External location is not a valid file system URI: " + location, e);
         }
+
+        // TODO (https://github.com/trinodb/trino/issues/17803) We cannot accept locations with double slash until all relevant Hive connector components are migrated off Hadoop Path.
+        // Hadoop Path "normalizes location", e.g.:
+        //  - removes double slashes (such locations are rejected),
+        //  - removes trailing slash (here ignored; foo/bar and foo/bar/ are treated as equivalent, and rejecting locations with trailing slash could pose UX issues)
+        //  - replaces file:///<local-path> with file:/<local-path> (here ignored).
+        if (validated.path().replaceFirst("/+$", "").contains("//")) {
+            throw new TrinoException(INVALID_TABLE_PROPERTY, "Unsupported location that cannot be internally represented: " + location);
+        }
```

Comment:
Rewrote the logic to accept locations such as `file:///tmp/6383092576615971655`, `file:///tmp/6383092576615971655/`, `s3://bucket/foo/`.

Reply:
as found in https://github.com/trinodb/trino/pull/17964 , two trailing slashes are not OK either.
i will  not add a test case for that, but will block such locations.
test case coming in the other PR